gateway/api_definition.go

```diff
@@ -626,6 +634,33 @@ func (a APIDefinitionLoader) replaceConsulSecrets(input *string) error {
 	return nil
 }
 
+func (a APIDefinitionLoader) replaceVaultSecrets(input *string) error {
+	if err := a.Gw.setUpVault(); err != nil {
+		return err
+	}
+
+	secret, err := a.Gw.vaultKVStore.(*kv.Vault).Client().Logical().Read(vaultSecretPath + prefixKeys)
```

Comment:
do we need thist type casting here? 
if it's absolutely required, can we do a type cast check here?

Reply:
resolved from internal conversation